src/types/mod.rs

```diff
@@ -358,9 +358,11 @@ impl<T: FromSql> FromSql for Vec<T> {
             _ => panic!("expected array type"),
         };
 
-        if try!(raw.read_i32::<BigEndian>()) != 1 {
-            return Err(Error::Conversion("array contains too many dimensions".into()));
-        }
+        match try!(raw.read_i32::<BigEndian>()) {
+            0 => return Ok(Vec::new()),
```

Comment:
Does a 0 dimensional array skip the rest of the metadata fields (has_nulls, member_oid, etc)? Does anything break if we just continue on when the number of dimensions is 0?